docs/intro/tutorial04.txt

```diff
@@ -303,19 +312,8 @@ views and use Django's generic views instead. To do so, open the
     def vote(request, question_id):
         ...  # same as above, no changes needed.
 
-We're using two generic views here:
```

Comment:
I appreciate that the content on this line was moved earlier in this document as I think it helps readability. Do you think it would be worthwhile to remind the reader of what each template does here too?

I can see such a reminder being helpful for readers who jump back to this specific section without reading any previous material. I know some people (myself included) will jump back to a specific section of a tutorial when they need to be reminded of how to accomplish some task in a new framework - in these cases, I think it would be helpful to have explanations of the generic views we are using in the examples (`ListView` and `DetailView`) near the examples themselves.

Reply:
Thank you for the review and comment.

I have made some changes towards your request, though I haven't made explicit mention of the views' goals to avoid duplication, since the same section that holds this code has the views' descriptions (the section is "Use generic views: Less code is better").

I extended the paragraph right after the code with some extra details. Then, the following paragraph mentions the views again, *and* it provides links to their definition. So I would think that's a good compromise? After my latest push, this piece of docs looks like this:

![image](https://github.com/django/django/assets/124304/cefa51ad-e8da-44a0-a164-09e81f59a12a)

What do you think?

Reply:
Thanks for the update!

That's very reasonable and LGTM.

Reply:
Thanks! Merging then.